src/mvc/AppPanel.java

```diff
@@ -48,11 +60,67 @@ public void display(){
     }
 
     public void actionPerformed(ActionEvent e) {
+        String cmmd = e.getActionCommand();
+        System.out.println(cmmd);
+        try {
+            switch (cmmd) {
+                case "Save": {
+                    String fName = Utilities.getFileName((String) null, false);
+                    ObjectOutputStream os = new ObjectOutputStream(new FileOutputStream(fName));
+                    os.writeObject(this.model);
+                    os.close();
+                    break;
+                }
+
+                case "Open": {
+
+                    if (Utilities.confirm("Are you sure? Unsaved changes will be lost!")) {
```

Comment:
Use Utilities.open() instead, it's safer and already implemented anyway (ditto with Utilities.save())